zkevm-circuits/src/state_circuit/lexicographic_ordering.rs

```diff
@@ -0,0 +1,278 @@
+use super::{N_LIMBS_ACCOUNT_ADDRESS, N_LIMBS_ID, N_LIMBS_RW_COUNTER};
+use crate::{
+    evm_circuit::{param::N_BYTES_WORD, witness::Rw},
+    gadget::is_zero::{IsZeroChip, IsZeroConfig, IsZeroInstruction},
+    util::Expr,
+};
+use eth_types::{Field, ToBigEndian};
+use halo2_proofs::{
+    circuit::Region,
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, VirtualCells},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use std::ops::Mul;
+
+// 2 limbs for tag, field_tag and id.
+// 10 limbs for address,
+// 16 limbs for storage key
+// 2 limbs for rw_counter
+// 30 limbs in total -> can fit into two field elements
+#[derive(Clone)]
+pub struct Config<F: Field> {
+    diff_1: Column<Advice>,
```

Comment:
could you rename `diff_1` `diff_2` more readable? such as diff + tag name or field name

Reply:
I can't think of better names for `diff_1` and `diff_2`.

To explain what they are:
We want to show that X > Y, where  A0, A1, ..., A29 are the 30 limbs of X and B0, B1, ..., B29 are the 30 limbs of Y, in big endian order. Let
    C0 = A0 - B0, 
    C1 = C0 << 16 + A1 - B1, 
    C2 = C1 << 16 + A2 - B2, 
    ...
    C14 = C13 << 16 + A14 - B14, 
and
    C15 = A15 - B15, 
    ...
    C29 = C28 << 16 + A29 - B29.

X > Y iff one of the following is true:
    1. one of C0, ..., C14 is non-zero and fits into 16 bits.
    2. all of C0, ..., C14 are 0 and one of C15, ..., C29 is non-zero and fits into 16 bits.
        (note that "all of C0, ..., C14 are 0" is equivalent to "C14 is 0")

The (diff_1, diff_2, diff_1_zero, diff_2_is zero) advice columns the constraints on them should be witnesses that prove that either 1. or 2. is true.

`diff_2` is constrained to be one of the 15 values C15, ..., C29. The limbs involved are the 13 least significant limbs of the storage key and the 2 limbs of the rw_counter. The limbs involved in`diff_1` are the remaining 15 (tag, field_tag, id, address, + 2 most significant limbs of the storage key).

I am open to suggestions for more descriptive names if you can think of any.